### PR TITLE
Introduce mechanism for ending engagement

### PR DESCRIPTION
**Pull request by dukhovnyi**
MOB-1031

### CODE REVIEW

File: GliaWidgets/Glia.swift [hunk 1]
```diff
@@ -285,4 +285,18 @@ public class Glia {
...
+    public func endEngagement(_ completion: @escaping (Result<Void, SalemoveError>) -> Void) {
+
+        interactor?.endSession(
+            success: {
+                completion(.success(()))
+            }, failure: {
+                completion(.failure($0))
+            }
+        )
+        onEvent?(.ended)
```
Comment (igorkravchenko) [inline GliaWidgets/Glia.swift:299]: While Interactor needs to be reworked in order to be mocked (this will be address some other time), for  now we can make sure that onEvent?(.ended) is called using unit test. Also in test we can make sure that nil is set to rootCoordinator (it has to become `internal` instead of `private`).

**Review by igorkravchenko — CHANGES_REQUESTED** [review]
One note regarding unit-tests, other than that - looks good.

File: GliaWidgets/Glia.swift (hunk 1 above)
Comment (dukhovnyi) [reply GliaWidgets/Glia.swift]: fixed 8743899cc91de4628974e20a922b41759ba1590c

**Review by igorkravchenko — APPROVED** [review]
Looks good! 👍
